modules/nf-cid/src/main/nextflow/data/cid/fs/CidPath.groovy

```diff
@@ -97,7 +106,7 @@ class CidPath implements Path, RealPathAware {
     /**
      * Finds the target path of a CID path
      **/
-    protected static Path findTarget(CidFileSystem fs, String filePath, String[] childs=[]) throws Exception{
+    protected static Path findTarget(CidFileSystem fs, String filePath, boolean resultsAsPath, String[] childs=[]) throws Exception{
```

Comment:
This method is quite large. Likely can be split in smaller unit e.g. 

```
if( object ) {
  findTargetForObject( .. _ 
}

if( resultsAsPath ) {
  .. etc
}
```

add add the corresponding tests

Reply:
Tests added as part of the getTargetPath tests